### PR TITLE
Portfolio: Local Storage.

### PR DESCRIPTION
**Pull request by maximoortelli**
_Portfolio-Local-Storage is an exercise in which I will preserve data in the browser of my portfolio. The goal here is to master all of the tools and best practices learned about in previous steps to use them in all Microverse projects and most likely in my future job as well._

### CODE REVIEW

File: js/localstorage.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const inputs = form.querySelectorAll('input, textarea');
+
+  // Function to save data to local storage
+  function saveData() {
+    const data = {};
+    inputs.forEach((input) => {
+      data[input.name] = input.value;
+    });
+    localStorage.setItem('formData', JSON.stringify(data));
+  }
```
Comment (christianbiring1) [inline js/localstorage.js]: - [x] As you may notice, when saving data in the localStorage and reloading the browser, the email is not saved. and while retrieving the data as you may see, the message is filling the email input. Kindly make sure you fix that.
image

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @maximoortelli , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHNAGES :recycle: 


## Highlights
- [x] Great work on saving and retrieving the data correctly in the localStorage :+1: 
- [x] Descriptive PR `title` and `summary` :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/localstorage.js (hunk 1 above)
Comment (maximoortelli) [reply js/localstorage.js]: Thank you very much for your review, I have already made all the suggested changes.👍

File: js/localstorage.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function localData() {
+  const localData = {
+    name: document.getElementById('savename').value,
+    email: document.getElementById('saveemail').value,
```
Comment (shahadat3669) [inline js/localstorage.js:4]: - [ ] 👋 Thank you for your efforts in addressing the previous review comments. However, it seems that a new error has arisen in your code. Previously, you had an email input field with the ID "saveemail," which you updated to "email" after the review feedback. Now, when attempting to add data, an error occurs: "Cannot read properties of null (reading 'value') at HTMLInputElement.localData (localstorage.js:4:48)."

[Portfolio-Complete (1).webm](https://github.com/maximoortelli/Portfolio-setup-and-mobile--first/assets/55840999/b402b9b4-e871-42f3-a3a0-af81271fa5ac)


Please resolve this issue 🙏, as it is preventing me from checking the updates made by the previous reviewer. By resolving this error, you'll be able to proceed with the further review of your changes. Keep up the great work, and happy coding! 💻🌟🚀

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
👋Hi @maximoortelli, This milestone is a [pair-programming activity](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/6_accessability.md#work-and-submission-mode).  For successful pair programming, both partners need to actively participate and contribute to the codebase. This will not only improve the quality of the project but also foster effective collaboration and learning. I recommend reaching out to your coding partner and coordinating your efforts to ensure that both of you are actively involved in the project.  Try to connect with your coding partner and make sure that both you and your partner are actively engaged in the project. Happy coding!

File: js/localstorage.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    message: document.querySelector('textarea').value,
+  };
+  localStorage.setItem('localData', JSON.stringify(localData));
+}
+
+const Inputs = document.querySelectorAll('#form input, #form textarea');
+Inputs.forEach((input) => {
+  input.addEventListener('input', localData);
+});
```
Comment (skyv26) [inline js/localstorage.js]: 👋🏻  Thank you for your efforts in addressing the previous review comments. However, a new error has arisen in your code. Previously, you had an email input field with the ID "save email," which you updated to "email" after the review feedback. Now, when attempting to add data, an error occurs: "Cannot read properties of null (reading 'value') at HTMLInputElement.localData (local storage.js:4:48)."


[256900114-b402b9b4-e871-42f3-a3a0-af81271fa5ac.webm](https://github.com/maximoortelli/Portfolio-setup-and-mobile--first/assets/66476329/d60ff576-41fa-4767-bb9a-b9109ca2071c)

Please resolve this issue 🙏, as it is preventing me from checking the updates made by the previous reviewer. By resolving this error, you'll be able to proceed with the further review of your changes. Keep up the great work, and happy coding! 💻🌟🚀

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @maximoortelli, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [OPTIONAL] Please make sure that you are following the project requirement and you should need to finish this activity as per pair-programming activity. But you did it alone, so please contact your partner to complete this activity.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/localstorage.js (hunk 3 above)
Comment (maximoortelli) [reply js/localstorage.js]: Thank you very much for your suggestion, I have made the requested changes, I thank you for your time.👍

**Review by rica213 — APPROVED** [review]
# 🟢 STATUS: APPROVED
Hi Maximo 
My name is Larissa. I am your code reviewer 🥰

Your project is complete! Thank you for implementing the changes requested by the previous reviewers 🛸 🛸 
Congratulations! 🎉 

## NOTE /!\ Next time you are having a pair-programming project, kindly do it with your coding partner. It will teach you on how to collaborate well with others, which is a highly sought-after skill by recruiters ❤️ 



## Highlights

- [x] No Linters errors ✔️ 
- [x] Descriptive PR ✔️ 
- [x] Correct Github flow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**